packages/adapters/bitcoin/src/connectors/WalletStandardConnector.ts

```diff
@@ -49,54 +52,146 @@ export class WalletStandardConnector extends ProviderEventEmitter implements Bit
       throw new Error('No account found')
     }
 
+    this.bindEvents()
+
     return account.address
   }
 
   async getAccountAddresses(): Promise<BitcoinConnector.AccountAddress[]> {
-    return Promise.resolve([])
+    const mappedAccounts = this.wallet.accounts.map<BitcoinConnector.AccountAddress>(acc => ({
+      address: acc.address,
+      purpose: 'payment',
+      publicKey: Buffer.from(acc.publicKey).toString('hex')
+    }))
+
+    return Promise.resolve(mappedAccounts)
   }
 
   async signMessage(params: BitcoinConnector.SignMessageParams): Promise<string> {
-    // BitcoinWalletStandard package is not exposing the signMessage feature
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    const feature = this.getWalletFeature('bitcoin:signMessage' as any)
+    const feature = this.getWalletFeature('bitcoin:signMessage')
 
     const account = this.wallet.accounts.find(acc => acc.address === params.address)
+
+    if (!account) {
+      throw new Error('Account not found')
+    }
+
     const message = new TextEncoder().encode(params.message)
     const response = (await feature.signMessage({ account, message }))[0]
 
-    // Should it be base64 encoded?
+    if (!response) {
+      throw new Error('No response from wallet')
+    }
+
     return Buffer.from(response.signature).toString('base64')
   }
 
   async signPSBT(
-    _params: BitcoinConnector.SignPSBTParams
+    params: BitcoinConnector.SignPSBTParams
   ): Promise<BitcoinConnector.SignPSBTResponse> {
-    return Promise.reject(new Error('Method not implemented.'))
+    const feature = this.getWalletFeature('bitcoin:signTransaction')
+
+    if (params.broadcast) {
+      throw new MethodNotSupportedError(
+        this.id,
+        'signPSBT',
+        'This wallet does not support broadcasting, please broadcast it manually or contact the development team.'
+      )
+    }
+
+    const inputsToSign = params.signInputs.map(input => {
+      const account = this.wallet.accounts.find(acc => acc.address === input.address)
+
+      if (!account) {
+        throw new Error(`Account with address ${input.address} not found`)
+      }
+
+      return {
+        account,
+        signingIndexes: [input.index],
+        sigHash: undefined
+      }
+    })
+
+    const response = (
+      await feature.signTransaction({
+        psbt: Buffer.from(params.psbt, 'base64'),
+        inputsToSign
+      })
+    )[0]
```

Comment:
We are checking just in the following line if it is undefined, but this should not happen. It is like that because `signTransaction` from the feature accepts a spread list of params and responds with an array of responses like `signTransaction(...args: SignTransactionRequest): SignTransactionResponse[]`.